x-pack/plugins/alerting/server/task_runner/action_scheduler/action_scheduler.ts

```diff
@@ -311,280 +127,51 @@ export class ActionScheduler<
       }
     }
 
-    if (!!bulkActionsResponse.length) {
-      for (const r of bulkActionsResponse) {
+    const actionsToNotLog: string[] = [];
+    if (!!bulkScheduleResponse.length) {
+      for (const r of bulkScheduleResponse) {
         if (r.response === ExecutionResponseType.QUEUED_ACTIONS_LIMIT_ERROR) {
-          ruleRunMetricsStore.setHasReachedQueuedActionsLimit(true);
-          ruleRunMetricsStore.decrementNumberOfTriggeredActions();
-          ruleRunMetricsStore.decrementNumberOfTriggeredActionsByConnectorType(r.actionTypeId);
-          ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
+          this.context.ruleRunMetricsStore.setHasReachedQueuedActionsLimit(true);
+          this.context.ruleRunMetricsStore.decrementNumberOfTriggeredActions();
+          this.context.ruleRunMetricsStore.decrementNumberOfTriggeredActionsByConnectorType(
+            r.actionTypeId
+          );
+          this.context.ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
             actionTypeId: r.actionTypeId,
             status: ActionsCompletion.PARTIAL,
           });
 
-          logger.debug(
+          this.context.logger.debug(
             `Rule "${this.context.rule.id}" skipped scheduling action "${r.id}" because the maximum number of queued actions has been reached.`
           );
 
-          delete logActions[r.id];
+          const uuid = r.uuid;
+          if (uuid) {
+            actionsToNotLog.push(uuid);
```

Comment:
Previously, we were creating a `logActions` record where the key was the `actionId` and the value was the information to pass to the event log in order to log the `execute-action` document. After adding some more unit tests in the `PerAlertActionScheduler`, I noticed that if we have multiple alerts for the same action or multiple actions that use the same connector, storing this information by the action ID would only allow one event log document to be written, so I switched to passing and using the action UUID instead.

Reply:
Not understanding why an action not having a `uuid` is treated differently here.  Based on this, but mainly the assignment of `actionsToLog` below, it seems like we won't log an event log doc for these?

Wondering if we could create a placeholder `uuid` when the action doesn't have one, just for the action scheduling bits here, so we wouldn't have to worry about it potentially being empty.  So I guess in at least `ExecutionResponseItem` and maybe `ExecuteOptions`, `uuid` wouldn't be optional.

Reply:
I created an issue about the optional typing and added comments about the optional typing here: https://github.com/elastic/kibana/pull/193807/commits/32beee0ea9e4c9275826e8e486fc852515ad4d86. I can add a placeholder UUID if you think that's better but I think that might cause more confusion in the future

Reply:
the comment is fine, no placeholder needed, thx!